### ISSUE TITLE
opcache_get_status() returns opcache_enabled = false

### ISSUE THREAD

**Issue by christoph-kluge** [brefphp/bref#439]
Hello everybody. I'm trying to measure and debug some performance issues and found out that opcache_enabled does return false. I'm kinda puzzled why this happens. Does someone have some hints for me?

My serverless.yml
```
functions:
    http:
        handler: public/lambda.php
        description: ''
        layers:
            - ${bref:layer.php-73}
        timeout: 3
        memorySize: 1024 # Overwrite the default memory size. Default is 1024
```

A simple handler which I'm using inside my app
```
class GetOpcacheHandler
{

    public function __invoke(Request $request)
    {
        $config = opcache_get_configuration();
        $status = opcache_get_status();

        return new JsonResponse([
            'is_cached' => opcache_is_script_cached(__FILE__),
            'config' => $config,
            'status' => $status,
        ]);
    }
}
```

And this is the response which I'm receiving:
```
{
	"is_cached": false,
	"config": {
		"directives": {
			"opcache.enable": true,
			"opcache.enable_cli": true,
			"opcache.use_cwd": true,
			"opcache.validate_timestamps": false,
			"opcache.validate_permission": false,
			"opcache.validate_root": false,
			"opcache.dups_fix": false,
			"opcache.revalidate_path": false,
			"opcache.log_verbosity_level": 1,
			"opcache.memory_consumption": 134217728,
			"opcache.interned_strings_buffer": 8,
			"opcache.max_accelerated_files": 10000,
			"opcache.max_wasted_percentage": 0.05,
			"opcache.consistency_checks": 0,
			"opcache.force_restart_timeout": 180,
			"opcache.revalidate_freq": 2,
			"opcache.preferred_memory_model": "",
			"opcache.blacklist_filename": "",
			"opcache.max_file_size": 0,
			"opcache.error_log": "",
			"opcache.protect_memory": false,
			"opcache.save_comments": true,
			"opcache.enable_file_override": false,
			"opcache.optimization_level": 2147401727,
			"opcache.lockfile_path": "\/tmp",
			"opcache.file_cache": "\/tmp",
			"opcache.file_cache_only": true,
			"opcache.file_cache_consistency_checks": true
		},
		"version": {
			"version": "7.3.7",
			"opcache_product_name": "Zend OPcache"
		},
		"blacklist": []
	},
	"status": {
		"opcache_enabled": false,
		"file_cache": "\/tmp",
		"file_cache_only": true
	}
}
```

**Comment by mnapoli** [comment]
That's interesting 🤔 

For reference, here is the `php.ini` for PHP functions: https://github.com/brefphp/bref/blob/master/runtime/layers/function/php.ini

It is different from HTTP applications, it uses the file cache only because every invocation of the function is in a new process. Can you confirm that you are not running a HTTP application here?

Maybe this is because of the file cache that this returns false? Does it return anything other than false when running on another machine?

**Comment by christoph-kluge** [comment]
Hi @mnapoli, thank you for your answer. 

> every incovation of the funcions is a new process 

was the answer to my problem. I'm running an http-server inside a long running process. Something like `php artisan serve` for laravel. With your hint I was able to reproduce the behavior locally. After some more testing I've found out that this is what I need and this exists already in your fpm example. So going to take this as a starting point 👍 

```
// test.php
do {
    require 'server.php';
} while (true);
```

```
// test-2.php
<?php

var_dump(
    opcache_is_script_cached(__FILE__),
    opcache_get_status()
);
```

Thank you so far!

**Comment by mnapoli** [comment]
:+1: thanks for the follow up.

I'm curious what you gain with a custom web server. In the lambda environment (and with the current state of the runtimes) everything would be faster with PHP-FPM I think.

**Comment by christoph-kluge** [comment]
Hi @mnapoli, quick update: 

I'm using Cloudfront@Edge with an Python-Lambda which forwards/aliases my requests to my own PHP Lambda. So my stack looks like this: `Cloudfront@Edge (Python) -> PHP Lambda (bref)`. Initially I wanted to use it for our API because we can leverage CloudFront to use Origin-Cache headers which works like a charm for specific endpoints but now I'm also using it for our frontend apps and our backoffice app. 

I've started new project which does all the transformations from a cloudfront-event to request/response objects which covers almost all cases which I need so far.

https://github.com/christoph-kluge/cloudfront-edge-php-adapter

This code runs since around 2 weeks in production. So far pretty stable. My response-times for the api are: minimum 3ms and avg: 15ms overall endpoints.

**Comment by christoph-kluge** [comment]
> I'm curious what you gain with a custom web server. In the lambda environment (and with the current state of the runtimes) everything would be faster with PHP-FPM I think.

Maybe I should state that it's not about performance. After Taylor Otwell did the presentation for  laravel vapor I didn't wanted to wait for the official release. So I decided to go down the road to run our api inside a lambda. I was courious how they have implemented it so I digged into CloudFront@Edge with an alias Lambda. Later I found out that vapor is based on API Gateway which was mentioned by [Mohamed Said in this article](https://divinglaravel.com/what-is-aws-lambda-and-how-laravel-vapor-uses-it).

Many many thanks to this project ❤️It did all the heavy lifting. Everything else was just a bit of AWS configuration and a little bit of code for the event-transformer..
